### PR TITLE
coverity: fix coverity issues on network, iotapi_event handler

### PR DESCRIPTION
**Pull request by pillip8282**
fix coverity issues on network, iotapi_event handler

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [9cd10d8eeaa7c2a5cafc66e1a9895a4ac4cc11d6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9cd10d8eeaa7c2a5cafc66e1a9895a4ac4cc11d6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0197b7537cde49e2775323c4c3c4b906b06d5c63] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0197b7537cde49e2775323c4c3c4b906b06d5c63] - Code Rule Check OK.

File: framework/src/wifi_manager/wifi_utils_lwnl80211.c [hunk 1]
```diff
@@ -251,8 +251,8 @@ int wifi_utils_callback_handler(int argc, char *argv[])
 				int ret = receive_scan_data(mqfd, scan_list);
 				if (ret == WIFI_UTILS_SUCCESS) {
 					g_cbk.scan_done(WIFI_UTILS_SUCCESS, scan_list, NULL);
+					free_scan_data(scan_list);
 				}
```
Comment (Hyunjun85) [inline framework/src/wifi_manager/wifi_utils_lwnl80211.c]: If WiFi scan has been failed, Won't  scan_list leaked?

**Pull request merged**